app/models/group.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+# OpenSCAP group
+class Group < ApplicationRecord
+  has_ancestry(primary_key_format: /\A[\w\-]+(\/[\w\-]+)*\z/)
+
+  belongs_to :benchmark, class_name: 'Xccdf::Benchmark'
+  has_many :rules, dependent: :destroy
+
+  validates :title, presence: true
+  validates :ref_id, uniqueness: { scope: %i[benchmark_id] }, presence: true
+  validates :description, presence: true
+  validates :benchmark_id, presence: true
+
+  def self.from_openscap_parser(op_group, benchmark_id: nil)
+    group = find_or_initialize_by(ref_id: op_group.id,
+                                    benchmark_id: benchmark_id)
+
+    group.assign_attributes(title: op_group.title,
+                         description: op_group.description,
+                         rationale: op_group.rationale,
+                         requires: op_group.requires,
+                         conflicts: op_group.conflicts,
+                         parent: Group.find_by(ref_id: op_group.parent_id),
```

Comment:
When we're importing N groups, this will fire N queries, this is heavily inefficient.

Reply:
Right, that's what I thought too... I'm going to try to think of a better way to do this